extensions/spotify-beta/src/search.tsx

```diff
@@ -0,0 +1,165 @@
+import { useState, useEffect } from "react";
+import { Action, ActionPanel, Grid, Icon, List, LocalStorage } from "@raycast/api";
+import { usePromise } from "@raycast/utils";
+import { useSearch } from "./hooks/useSearch";
+import { View } from "./components/View";
+import { ArtistsSection } from "./components/ArtistsSection";
+import { AlbumsSection } from "./components/AlbumsSection";
+import { TracksSection } from "./components/TracksSection";
+import { PlaylistsSection } from "./components/PlaylistsSection";
+import { debounce } from "./helpers/debounce";
+import { ShowsSection } from "./components/ShowsSection";
+import { EpisodesSection } from "./components/EpisodessSection";
+
+const filters = {
+  all: "All",
+  artists: "Artists",
+  tracks: "Songs",
+  albums: "Albums",
+  playlists: "Playlists",
+  shows: "Podcasts",
+  episodes: "Episodes",
+};
+
+type FilterValue = keyof typeof filters;
+
+function SearchCommand() {
+  const {
+    data: recentSearchesData,
+    isLoading: recentSearchIsLoading,
+    revalidate: recentSearchRevalidate,
+  } = usePromise(() => LocalStorage.getItem<string>("recent-searches"));
+  const [searchText, setSearchText] = useState("");
+  const [searchFilter, setSearchFilter] = useState<FilterValue>("all");
+  const { searchData, searchIsLoading } = useSearch({
+    query: searchText,
+    limit: 32,
+    options: { keepPreviousData: true },
+  });
+
+  const recentSearches: string[] = recentSearchesData ? JSON.parse(recentSearchesData) : [];
+
+  useEffect(() => {
+    if (searchText.length > 3 && recentSearches.includes(searchText.trim()) === false && searchIsLoading === false) {
+      const addSearchToStorage = debounce(async () => {
+        LocalStorage.setItem("recent-searches", JSON.stringify([searchText, ...recentSearches]));
```

Comment:
Why not set the search to the recent searches once Spotify has returned valid data? That would prevent searches from returning no results to be saved.

Reply:
It makes sense but I belive this could be solved without using a `debounce` function. There's a [`throttle` option](https://developers.raycast.com/api-reference/user-interface/list#props) on the List component that could be used here. I'm wondering if it's not possible to achieve the same by using the `throttle` option and running an effect once the search data is not loading and you have results 🤔

Reply:
Yeah, Im pretty sure I tried that. This was a couple of weeks ago so happy to try it again though!